content/reference/cd-supporting-new-ci.md

```diff
@@ -0,0 +1,79 @@
+1. Detecting additional environment variables for a specific CI system
+
+In the [`pulumi/pulumi`](https://github.com/pulumi/pulumi) repo the [`pkg/util/ciutil`](https://github.com/pulumi/pulumi/tree/master/pkg/util/ciutil) contains all of the logic necessary for the above. Every stack `preview` or `update` calls the functions exported from this package.
+
+### Detecting The CI System
+
+- Add an entry to the local `map` in [`detect.go`](https://github.com/pulumi/pulumi/blob/master/pkg/util/ciutil/detect.go) where all of the CI systems are registered.
+- When you add the entry, you have two options, either use the `baseCI` as the CI System implementation or add a specific implementation that overrides the `DetectVars` function. More on this later in the next section.
+
+For example, here's the `AppVeyor` entry.
+
+```go
+AppVeyor: baseCI{
+		Name:            AppVeyor,
+		EnvVarsToDetect: []string{"APPVEYOR"},
+},
+```
+
+The `EnvVarsToDetect` is used by the `IsCI()` in [`systems.go`](https://github.com/pulumi/pulumi/blob/master/pkg/util/ciutil/systems.go), which iterates through the environment variables that a certain CI system is known to set in its build agents. Some CI systems set specific _values_ in certain environment variables, and in such cases you should use `EnvValuesToDetect`. An example for the latter is `Codeship`. See its entry in the `detectors` map.  
+
+### Detecting Additional Metadata About A CI Build
+
+A CI build could have been triggered by a PR or a push build. In the case of the former, the linked PR number is useful so that the Pulumi Console can add a link to it in the source repo of your stack. Note that the detection is done on a best-effort basis. We support linking to the following Git-based version control systems:
+
+- Atlassian Bitbucket
+- Azure DevOps
+- GitHub
+- GitLab
+
+All of the above source control systems have a concept of a PR, and PR builds, as well as push builds. 
+
+- Add a new file to the [`pkg/util/ciutil`](https://github.com/pulumi/pulumi/tree/master/pkg/util/ciutil) folder with the name of the CI system for which you are adding support.
+- Define a new struct for the CI system and add `baseCI` to its definition. Refer to any of other pre-existing implementations, such as the [`travis.go`](https://github.com/pulumi/pulumi/blob/master/pkg/util/ciutil/travis.go) file.
+- Implement the `DetectVars()` function. This is the function where you should construct a new instance of the `Vars` struct and set its properties accordingly.
+
+That's it! Send us a new [PR](https://github.com/pulumi/pulumi/pulls) in the [`pulumi/pulumi`](https://github.com/pulumi/pulumi) repo with your addition. We would love to help you get that merged as soon as possible.
+
+## Using A Fallback
+
+If the CI system you are using is not current detected by Pulumi. You can set the following env vars, as a fallback, yourself. The CI detection code will assume a "Generic CI" system name.
```

Comment:
current => currently

And:

“Pulumi. You” => “Pulumi, you”

?

Reply:
Done. I also rephrased the sentence a bit.